src/main/assets/js/opening-hours.ts

```diff
@@ -0,0 +1,88 @@
+import $ from 'jquery';
+
+const formId = '#openingTimesForm';
+const tabId = '#openingTimesTab';
+const newOpeningTimeHeadingId = '#newOpeningHoursHeading';
+const openingTimesContentId = '#openingTimesContent';
+const deleteBtnClass = 'deleteOpeningTime';
+const addOpeningTimesBtnName = 'addOpeningTime';
+const typeSelectName = 'type_id';
+const hoursInputName = 'hours';
+
+const getInputName = (name: string, index: number): string => {
+  return `opening_times[${index}][${name}]`;
+};
+
+// Rename the input fields so that the index values are in order,
+// which affects the order when the form is posted.
+const renameFormElements = (): void => {
+  $(`${tabId} select[name$="[${typeSelectName}]"]`)
+    .attr('name', idx => getInputName(typeSelectName, idx))
+    .attr('id', idx => 'description-' + idx);
+  $(`${tabId} input[name$="[${hoursInputName}]"]`)
+    .attr('name', idx => getInputName(hoursInputName, idx))
+    .attr('id', idx => 'hours-' + idx);
+};
+
+$(() => {
+  if($(tabId).length > 0) {
+
+    // GET the opening-hours data for a court
+    const slug = $('#slug').val();
+    $.ajax({
+      url: `/courts/${slug}/opening-times`,
+      method: 'get',
+      success: (res) => {
+        $(openingTimesContentId).html(res);
+      },
+      error: (jqxhr, errorTextStatus, err) =>
+        console.log(`GET opening times failed: ${errorTextStatus} ${err}`)
+    });
+
+    // POST the opening-hours form for a court
+    $(formId).on('submit', e => {
+      e.preventDefault();
+
+      const url = $(e.target).attr('action');
+      $.post(url, $(e.target).serialize())
```

Comment:
A small one, so ignore if its silly

1) The endpoint on the API is a put, and we make a call to it as a put in the src/main/assets/js/opening-hours.ts file. Is it worth declaring the other stuff as puts as well to make it more consistant?

Reply:
I think this is a good idea. The code has been updated to use PUT.